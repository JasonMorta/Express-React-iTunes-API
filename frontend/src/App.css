.App {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(0.625rem + 2vmin);
  color: white;
  position: relative;
  display: flex;
  background-image: url('./ptn1.gif');
}

body {
  padding-right: 0px !important;
}

.iTune-container {
  max-width: 50rem;
  /* height: 56.25rem; */
  position: relative;
  background-color: #F5F5F5;
  padding: 0.625rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 0.625rem;

}

.input-group {
  width: 75%;
  align-items: center;
  margin: auto;
}

.btn-check:focus+.btn-outline-secondary, .btn-outline-secondary:focus {
  box-shadow: inset 0 0 0 0.25rem rgb(108 117 125 / 50%);
}

.header-content {
  display: flex;
  justify-content: center;
}

.header-content img {
  margin: 1.25rem;
  margin-bottom: 0rem;
}

.navigation-container {
  color: black;
}

.mb-3 {
  position: relative;
}

.music-thead {
  margin: auto;
}

.fav-counter {
  background-color: #f44336;
  font-size: 0.9375rem;
  border-radius: 50%;
  width: 1.25rem;
  height: 1.25rem;
  color: white;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 2;
}

.favBtn-container {
  display: flex;
  justify-content: flex-end;
  position: absolute;
  right: 0.75rem;
  top: 35%;
}

.favorites {
  border-radius: 50%;
  padding: 0.625rem;
  background-color: #8f15ad;
  border: 0px;
  font-size: small;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2.5rem;
  height: 2.5rem;
  margin: 0.3125rem;
  z-index: 1;
}

tbody img {
  margin: 0.1875rem;
}

tbody tr {
  vertical-align: middle;
}

tbody tr:hover {
  background-image: url(./bg.png);

}

.tab-pane {
  overflow: auto;
width: inherit;
}

::-webkit-scrollbar {
  width: 0.5rem;
}
/* Track */
::-webkit-scrollbar-track {
  box-shadow: inset 0 0 5px grey; 
  border-radius: 0.625rem;
}
 
/* Handle */
::-webkit-scrollbar-thumb {
  background: #8f15ad; 
  border-radius: 0.625rem;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #641179; 
}


.tab-content>.active {
  box-shadow: inset 0px 0px 2.75rem -1.4375rem #0096af;
  border-radius: 0.3125rem;
  background: linear-gradient(320deg, #a2c5ec, #e8e8ea);
  background-size: 400% 400%;
  height: 29.75rem;

  -webkit-animation: AnimationName 15s ease infinite;
  -moz-animation: AnimationName 15s ease infinite;
  animation: AnimationName 15s ease infinite;
}

@-webkit-keyframes AnimationName {
  0% {
    background-position: 4% 0%
  }

  50% {
    background-position: 97% 100%
  }

  100% {
    background-position: 4% 0%
  }
}

@-moz-keyframes AnimationName {
  0% {
    background-position: 4% 0%
  }

  50% {
    background-position: 97% 100%
  }

  100% {
    background-position: 4% 0%
  }
}

@keyframes AnimationName {
  0% {
    background-position: 4% 0%
  }

  50% {
    background-position: 97% 100%
  }

  100% {
    background-position: 4% 0%
  }
}

.modalBtn {
  padding: 0.625rem !important;
  background-color: #8f15ad !important;
  border: 0px !important;
}

.modal-body {
  overflow: auto;
  height: 60vh;
}

.modalBtn:focus,
.favorites:focus {
  background-color: #9c27b0 !important;
  border-color: #9c27b0 !important;
  box-shadow: 0 0 0 0.25rem rgb(156 39 176 / 45%) !important;

}

.modalBtn:active,
.favorites:active {
  background-color: #008596 !important;
}

.modalBtn:hover,
.favorites:hover {
  background-color: #be45d2 !important;
}

.delete-block {
  fill: #ce93d8;
  text-align: center;
}

.delete-block svg:hover {
  fill: red;
}

.card {
  font-size: medium;
  box-shadow: 0.0625rem 0.0625rem 0.1875rem #808080f7;
  transition: .5s;
  padding: 0.1875rem;
}

.card-body {
  padding: .3rem .3rem;
  font-size: small;
}

.card-title {
  font-size: medium !important;
}

.card::after {
  content: "";
  width: 0%;
  height: 0.1875rem;
  transition: 1s;
  margin: auto;
}

.card:hover::after {
  content: "";
  width: 100%;
  height: 0.1875rem;
  background-color: #9c27b0;
  transition: .5s;
}


.music-videos-card {
  max-height: 29.0625rem;
  width: 100%;
  margin: 0;
}

.movie-card {
  max-height: 29.0625rem;
  width: 100%;
  margin: 0;
  padding-bottom: 1.875rem;
}

.movie-card .card {
  max-height: none;
}

.tv-card {
  margin-bottom: .1rem;
  border-bottom: 1px solid #8080802b;
}

.nav-link {

  color: #2196f3;
}

.loader {
  font-size: 0.625rem;
  margin: 0px;
  overflow: hidden;
  text-indent: -9999em;
  width: 1.5rem;
  height: 1.5rem;
  border-radius: 50%;
  background: #48db8f;
  background: -moz-linear-gradient(left, #48db8f 10%, rgba(72, 219, 143, 0) 42%);
  background: -webkit-linear-gradient(left, #48db8f 10%, rgba(72, 219, 143, 0) 42%);
  background: -o-linear-gradient(left, #48db8f 10%, rgba(72, 219, 143, 0) 42%);
  background: -ms-linear-gradient(left, #48db8f 10%, rgba(72, 219, 143, 0) 42%);
  background: linear-gradient(to right, #48db8f 10%, rgba(72, 219, 143, 0) 42%);
  position: relative;
  -webkit-animation: load3 .5s infinite linear;
  animation: load3 .5s infinite linear;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.loader:before {
  width: 50%;
  height: 50%;
  background: #48db8f;
  border-radius: 100% 0 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}

.loader:after {
  background: #ffffff;
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

@-webkit-keyframes load3 {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes load3 {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.search-heading {
  color: #0096af;
  display: flex;
  justify-content: center;
}

.form-control:focus {
  color: #212529;
  background-color: #f5f5f5;
  border-color: #0096af;
  outline: 0;
  box-shadow: inset 0 0 0 0.19rem rgb(0 150 175 / 28%);
}
