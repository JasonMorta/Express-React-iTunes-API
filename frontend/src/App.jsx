import "bootstrap/dist/css/bootstrap.min.css";
import Button from "react-bootstrap/Button";
import NavigationTabs from "./components/NavigationTabs";
// import Music from './components/Music';
import SearchFor from "./components/SearchFor";
import logo from "./logo.png";
import "./App.css";
import "./fonts.css";
import FavoritesModal from "./components/FavoritesModal";

import React, { Component } from "react";

export default class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      itunesSongs: [],
      itunesSongsObject: [],
      itunesMovies: [],
      itunesMoviesObject: [],
      itunesMusicVideos: [],
      itunesMusicVideosObject: [],
      itunesTvShows: [],
      itunesTvShowsObject: [],
      itunesPodcast: [],
      itunesPodcastObject: [],
      tabValue: "1",
      empty: [],
      addPlayer: false,
      findMedia: "",
      modalShow: false,
      changeFavIcon: false,
      inputValue: "",
      media: "all",
      favoritesArray: [],
      favCounter: 0,
      refreshState: false,
      loading: false,
    };
  }


  

  //SEARCH button
  searchBtn = () => {
    //Validation: The user must enter a value to get results from search.
    if (this.state.inputValue === "") {
      console.log("inputValue: " + typeof this.state.inputValue.length);
      alert("Please enter a value");
    } else {
      this.setState({
        itunesSongs: [],
        itunesSongsObject: [],
        itunesMovies: [],
        itunesMoviesObject: [],
        itunesMusicVideos: [],
        itunesMusicVideosObject: [],
        itunesTvShows: [],
        itunesTvShowsObject: [],
        itunesPodcast: [],
        itunesPodcastObject: [],
        isFav: false,
        changeHeart: "",
        loading: true,
      });
      console.log("inputValue: " + this.state.inputValue);
      //Get Songs
      fetch("https://itunedbackend.onrender.com/songs", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          name: this.state.inputValue,
        }),
      })
        .then((res) => {
          if (!res.ok) {
            throw new Error(`Request failed with status: ${res.status}`);
          }
          return res.json();
        })
        .then((response) => {
          console.log('response', response)
          this.setState(
            {
              itunesSongs: response,
              itunesSongsObject: response.results,
              loadingDataBtn: false,
            },
            () => {}
          );
        })
        .catch((error) => console.log("Error:", error));

      //Get Movies
      fetch("https://itunedbackend.onrender.com/movie", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          name: this.state.inputValue,
        }),
      })
        .then((res) => res.json())
        .then((songs) => {
          this.setState({
            itunesMovies: songs,
            itunesMoviesObject: songs.results,
            loadingDataBtn: false,
          });
        })
        .catch((error) => console.log("Error:", error));

      //Get Music Video
      fetch("/musicVideos", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          name: this.state.inputValue,
        }),
      })
        .then((res) => res.json())
        .then((response) => {
          this.setState({
            itunesMusicVideos: response,
            itunesMusicVideosObject: response.results,
            loadingDataBtn: false,
          });
        })
        .catch((error) => console.log("Error:", error));
console.count("rerender")
      //Get podcasts
      fetch("https://itunedbackend.onrender.com/TVshows", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          name: this.state.inputValue,
        }),
      })
        .then((res) => res.json())
        .then((response) => {
          this.setState({
            itunesTvShows: response,
            itunesTvShowsObject: response.results,
            loadingDataBtn: false,
          });
        })
        .catch((error) => console.log("Error:", error));

      //Get podcasts
      fetch("https://itunedbackend.onrender.com/podcasts", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          name: this.state.inputValue,
        }),
      })
        .then((res) => res.json())
        .then((response) => {
          this.setState(
            {
              itunesPodcast: response,
              itunesPodcastObject: response.results,
              loading: false,
            },
            () => {}
          );
        })
        .catch((error) => console.log("Error:", error));
    }
  }; //end search

  //Get input field values for search
  storeUserInput = (e) => {
    this.setState({ inputValue: e.target.value }, () => {});
  };

  //Updated the favorite counter bubble
  updateFavCounter = () => {
    if (this.state.favoritesArray.length > -1) {
      this.setState({
        favCounter: this.state.favoritesArray.length,
      });
    } else {
      console.log("No update");
    }
  };

  //this is need to rerender the UI when a childe component changes it's own state, but not updating the UI.
  refreshState = () => {
    this.setState({
      refreshState: true,
      favCounter: this.state.favoritesArray.length,
    });
  };

  render() {
    return (
      <div className="App">
        <div className="iTune-container">
          <p className="my-id">
            CREATED BY:
            <a
              href="https://github.com/JasonMorta?tab=repositories"
              target="_blank"
              rel="noreferrer"
              title="GitHub Profile"
            >
              JASON MORTA
            </a>
          </p>
          <div className="header-content">
            <img src={logo} alt="design-logo" />
          </div>
          <div>
            <SearchFor
              loading={this.state.loading}
              inputValue={this.state.inputValue}
              storeUserInput={this.storeUserInput.bind(this)}
              searchBtn={this.searchBtn.bind(this)}
            />
          </div>
          <div className="favBtn-container">
            {/* This is the red favorites counter bubble. it will only appear if there are favs */}
            {this.state.favCounter > 0 ? (
              <span className="fav-counter">{this.state.favCounter}</span>
            ) : (
              <></>
            )}
            {/* This button that opens the favorites list */}
            <Button
              variant="primary"
              onClick={() =>
                this.setState({
                  modalShow: true,
                })
              }
              className="favorites"
            >
              Favs
            </Button>
            {/* The favorites List component */}
            <FavoritesModal
              favoritesArray={this.state.favoritesArray}
              show={this.state.modalShow}
              onHide={() =>
                this.setState({
                  modalShow: false,
                })
              }
              refreshState={this.refreshState.bind(this)}
            />
          </div>
          {/* All Tabs */}
          <div className="navigation-container">
            <NavigationTabs
              updateFavCounter={this.updateFavCounter.bind(this)}
              songList={this.state.itunesSongsObject}
              podcastList={this.state.itunesPodcastObject}
              movieList={this.state.itunesMoviesObject}
              musicVideosList={this.state.itunesMusicVideosObject}
              tvShows={this.state.itunesTvShowsObject}
              favoritesArray={this.state.favoritesArray}
              favoritesArrayLength={this.state.favoritesArray}
            />
          </div>

          <div className="footer-content"></div>
        </div>
      </div>
    );
  }
}
