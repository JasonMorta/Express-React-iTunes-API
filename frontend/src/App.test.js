import React from 'react';
import App from '../src/App.jsx'

import renderer from 'react-test-renderer';


it('renders correctly', () => {
   const tree = renderer
   .create(<App/>)
   expect(tree).toMatchSnapshot();
   });