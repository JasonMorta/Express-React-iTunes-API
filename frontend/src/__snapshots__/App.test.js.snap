// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="iTune-container"
  >
    <div
      className="header-content"
    >
      <img
        alt="design-logo"
        src="logo.png"
      />
    </div>
    <div>
      <div
        className="search-container"
      >
        <div
          className="search-heading"
        >
          <p>
            Enter a name to search for
          </p>
        </div>
        <div
          className="search-bar"
        >
          <div
            className="mb-3 input-group has-validation"
          >
            <input
              aria-describedby="basic-addon2"
              aria-label="Search.."
              className="form-control"
              onChange={[Function]}
              placeholder="Search.."
              required={true}
              value=""
            />
            <button
              className="btn btn-outline-secondary"
              disabled={false}
              id="button-addon2"
              onClick={[Function]}
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="favBtn-container"
    >
      <button
        className="favorites btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Favs
      </button>
    </div>
    <div
      className="navigation-container"
    >
      <ul
        className="mb-3 nav nav-tabs"
        onKeyDown={[Function]}
        role="tablist"
      >
        <li
          className="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-tab1"
            aria-selected={true}
            className="nav-link active"
            data-rr-ui-event-key="tab1"
            disabled={false}
            id="uncontrolled-tab-example-tab-tab1"
            onClick={[Function]}
            role="tab"
            type="button"
          >
            Music
          </button>
        </li>
        <li
          className="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-tab2"
            className="nav-link"
            data-rr-ui-event-key="tab2"
            disabled={false}
            id="uncontrolled-tab-example-tab-tab2"
            onClick={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Music Videos
          </button>
        </li>
        <li
          className="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-tab3"
            className="nav-link"
            data-rr-ui-event-key="tab3"
            disabled={false}
            id="uncontrolled-tab-example-tab-tab3"
            onClick={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Movies
          </button>
        </li>
        <li
          className="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-tab4"
            className="nav-link"
            data-rr-ui-event-key="tab4"
            disabled={false}
            id="uncontrolled-tab-example-tab-tab4"
            onClick={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Podcasts
          </button>
        </li>
        <li
          className="nav-item"
          role="presentation"
        >
          <button
            aria-controls="uncontrolled-tab-example-tabpane-tab5"
            className="nav-link"
            data-rr-ui-event-key="tab5"
            disabled={false}
            id="uncontrolled-tab-example-tab-tab5"
            onClick={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            TV Shows
          </button>
        </li>
      </ul>
      <div
        className="tab-content"
      >
        <div
          aria-labelledby="uncontrolled-tab-example-tab-tab1"
          className="fade tab-pane active show"
          id="uncontrolled-tab-example-tabpane-tab1"
          role="tabpanel"
          style={
            Object {
              "fontSize": "1rem",
              "overflow": "auto",
            }
          }
        >
          <div
            className="table-responsive"
          >
            <table
              className="table table-striped table-bordered table-hover"
              onClick={[Function]}
            >
              <thead>
                <tr
                  className="music-thead"
                >
                  <th>
                    #
                  </th>
                  <th>
                    Name
                  </th>
                  <th>
                    Artist
                  </th>
                  <th>
                    Album
                  </th>
                  <th>
                    Time
                  </th>
                  <th>
                    Fav
                  </th>
                </tr>
              </thead>
              <tbody
                style={
                  Object {
                    "fontSize": "1rem",
                  }
                }
              />
            </table>
          </div>
        </div>
        <div
          aria-labelledby="uncontrolled-tab-example-tab-tab2"
          className="fade tab-pane"
          id="uncontrolled-tab-example-tabpane-tab2"
          role="tabpanel"
        >
          <div
            className="g-4 music-videos-card row row-cols-md-4 row-cols-2"
          />
        </div>
        <div
          aria-labelledby="uncontrolled-tab-example-tab-tab3"
          className="fade tab-pane"
          id="uncontrolled-tab-example-tabpane-tab3"
          role="tabpanel"
        >
          <div
            className="g-4 movie-card row row-cols-md-4 row-cols-2"
          />
        </div>
        <div
          aria-labelledby="uncontrolled-tab-example-tab-tab4"
          className="fade tab-pane"
          id="uncontrolled-tab-example-tabpane-tab4"
          role="tabpanel"
          style={
            Object {
              "fontSize": "1rem",
              "overflow": "auto",
            }
          }
        >
          <table
            className="table table-striped table-bordered table-hover"
          >
            <thead>
              <tr
                className="music-thead"
              >
                <th>
                  #
                </th>
                <th>
                  Name
                </th>
                <th>
                  Provider
                </th>
                <th>
                  Podcasts
                </th>
                <th>
                  Time
                </th>
                <th>
                  Fav
                </th>
              </tr>
            </thead>
            <tbody
              style={
                Object {
                  "fontSize": "1rem",
                }
              }
            />
          </table>
        </div>
        <div
          aria-labelledby="uncontrolled-tab-example-tab-tab5"
          className="fade tab-pane"
          id="uncontrolled-tab-example-tabpane-tab5"
          role="tabpanel"
        >
          <div
            className="g-4 movie-card tv-card row row-cols-md-4 row-cols-2"
          />
        </div>
      </div>
    </div>
    <div
      className="footer-content"
    />
  </div>
</div>
`;
